### PR TITLE
Adds Docker Compose and Poetry toml and lock files

### PR DESCRIPTION
**Pull request by jermnelson**
Enable better dependency management with [Poetry](https://python-poetry.org/). Adds new `docker-compose.yaml` for running an Airflow environment.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,24 @@
 # folio-airflow
 Airflow DAGS for migrating ILS data into FOLIO
+
+## Running Locally with Docker
+Based on the documentation, [Running Airflow in Docker](https://airflow.apache.org/docs/apache-airflow/stable/start/docker.html). 
+
+> **NOTE** Make sure there is enough RAM available locally for the 
+> docker daemon, we recommend at least 5GB. 
+
+1. Clone repository `git clone https://github.com/LD4P/ils-middleware`
```
Comment (jgreben) [inline README.md:10]: Should this be `sul-dlss/folio-airflow`?

**Review by jgreben — CHANGES_REQUESTED** [review]
In the readme maybe a copy/paste typo?

**Comment by jgreben** [comment]
@jermnelson please check the PR when you get a chance. Thanks!

File: README.md (hunk 1 above)
Comment (jermnelson) [reply README.md]: Good catch! Just fixed with latest commit.

**Review by jgreben — APPROVED** [review]

**Pull request merged**